pkg/test/framework/components/echo/echotest/run.go

```diff
@@ -74,11 +77,15 @@ func (t *T) Run(testFn oneToOneTest) {
 //      ingr.CallWithRetryOrFail(...)
 //    })
 func (t *T) RunFromClusters(testFn oneClusterOneTest) {
-	t.fromEachCluster(t.rootCtx, func(ctx framework.TestContext, c cluster.Cluster) {
-		t.toEachDeployment(ctx, func(ctx framework.TestContext, dstInstances echo.Instances) {
-			t.setupDst(ctx, dstInstances)
-			testFn(ctx, c, dstInstances)
-		})
+	t.toEachDeployment(t.rootCtx, func(ctx framework.TestContext, dstInstances echo.Instances) {
+		t.setupPair(ctx, nil, echo.Services{dstInstances})
+		if len(ctx.Clusters()) == 1 {
```

Comment:
why do we need to check for single cluster?

Reply:
It's mostly aesthetic, we do this elsewhere, and avoid an intermediate subtest when not necessary.

Reply:
It's a tradeoff of aesthetics for the test output and aesthetics of the code. I'd rather favor simple code TBH. Having checks for cluster count sprinkled around might cause confusion

Reply:
IMO time spent debugging tests >>> time spent reading the test framework code. This is in the framework, not the individual tests.

Some of the current tests are pretty ridiculous and are nested 15 levels deep making it hard to debug

Reply:
My concern is more about someone looking around at other examples for writing a test and wondering if they're supposed to have specialized single-cluster code too. I understand that we have a lot of tests that are complex, and we should focus on reducing the complexity in those cases. 

But looking at the diff, it appears that the previous code didn't have this special case .. it was added in this PR. Seems that there would be no change in debuggability from the previous code if we removed it. Or am I missing something?

Reply:
This isn't a test though, its the internals of the test framework. I agree a test should not have anything like this in it

Reply:
FWIW the original `Run` + internal `fromEachCluster` had this – I didn't add the `RunFromClusters` but I tried to make each of the `RunX` methods more consistent.

Reply:
ah my mistake ... it's fine. Please ignore.